test/readMPAS.f90

```diff
@@ -30,6 +30,137 @@ function epsilonDiff(a,b) result(isSame)
   endif
 end function
 
+
+subroutine rebuildTests(mpMesh, numMPs, mp2Elm, isMPActive)
+    use :: polympo
+    use iso_c_binding
+    implicit none
+    type(c_ptr):: mpMesh
+    integer :: numMPs, i, numMPsLarger
+    integer, dimension(:), pointer :: mp2Elm, addedMPMask, isMPActive, mp2ElmFromPMPO
+    integer, dimension(:), pointer :: mp2ElmLarger, addedMPMaskLarger, mp2ElmFromPMPOLarger, isMPActiveLarger
+    integer, parameter :: MP_ACTIVE = 1
+    integer, parameter :: MP_INACTIVE = 0
+    integer, parameter :: MP_DELETE_ELM_ID = -1
+
+    ! TEST: adding MP
+    ! Necessary pre-conditions for test
+    call assert(numMPs >= 4, "not enough MPs for test")
+    call assert(isMPActive(4) == MP_INACTIVE, "mp2Elm = 4 is active")
+    ! PREPARE DATA
+    allocate(addedMPMask(numMPs))
+    addedMPMask = MP_INACTIVE
+    isMPActive(4) = MP_ACTIVE
+    mp2Elm(4) = 7
+    addedMPMask(4) = MP_ACTIVE
+    ! Rebuild MPs
+    call polympo_rebuildMPs(mpMesh,numMPs,c_loc(mp2Elm),c_loc(addedMPMask))
+    ! Test values
+    allocate(mp2ElmFromPMPO(numMPs))
+    mp2ElmFromPMPO = MP_DELETE_ELM_ID
+    call polympo_getMPCurElmID(mpMesh,numMPs,c_loc(mp2ElmFromPMPO))
+    do i = 1, numMPs
+        if (isMPActive(i) == MP_ACTIVE) then
+            call assert(mp2Elm(i) .eq. mp2ElmFromPMPO(i), "wrong element ID for i'th MP after rebuild")
+        endif
+    end do
+
+    ! TEST: deleting two MPs
+    ! Necessary pre-conditions for test
+    call assert(numMPs >= 4, "not enough MPs for test")
+    call assert(isMPActive(1) == MP_ACTIVE, "mp2Elm = 1 not active")
+    call assert(isMPActive(4) == MP_ACTIVE, "mp2Elm = 4 not active")
+    ! PREPARE DATA
+    isMPActive(1) = MP_INACTIVE
+    isMPActive(4) = MP_INACTIVE
```

Comment:
How do the assertions on lines 71 and 72 above matter if we change the settings for these two MPs?

Reply:
Onkar wanted the assertions because it makes the code easier to read for our customers. Also we want make sure that the test is changing something. So we check the values are different before and after.

Reply:
Hmm.  OK.  If it is a check on the results of the prior test (or user input) then it makes sense, but then I'd suggest moving it up into the section of code for that test to make it clearer.